torch/_tensor_str.py

```diff
@@ -599,11 +599,22 @@ def indented_str(s, indent):
 
     # Use inp here to get the original grad_fn and not the one generated by the forward grad
     # unpacking.
-    if inp.grad_fn is not None:
-        name = type(inp.grad_fn).__name__
-        if name == "CppFunction":
-            name = inp.grad_fn.name().rsplit("::", 1)[-1]
-        suffixes.append(f"grad_fn=<{name}>")
+    grad_fn_name = None
+    try:
+        grad_fn = inp.grad_fn
+    except RuntimeError:
+        # Accessing the grad_fn calls rebasing logic which would cause an error
+        # if that tensor is a view created in no-grad mode modified in-place in
+        # no-grad mode. See: https://github.com/pytorch/pytorch/issues/99968
+        grad_fn_name = "Undefined"
+
+    if grad_fn_name is None and grad_fn is not None:
+        grad_fn_name = type(grad_fn).__name__
+        if grad_fn_name == "CppFunction":
+            grad_fn_name = grad_fn.name().rsplit("::", 1)[-1]
```

Comment:
You can leave this in the try so that you don't need the condition?

Reply:
If grad_fn.name() fails or doesn't return a string, wouldn't we rather have a legit error? Also, rsplit is a no-op anyway if we have a custom autograd Function, so we didn't need the condition in the first place, though maybe I prefer this being explicit.

Reply:
Ok.